rocAL/rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -22,66 +22,80 @@
-                self.labels_tensor = self.labels.reshape(-1, self.bs, self.loader._numOfClasses)
+        if self.output_list is None:  # Checking if output_list is empty and initializing the buffers
+            self.output_list = []
+            for i in range(len(self.output_tensor_list)):
+                self.dimensions = self.output_tensor_list[i].dimensions()
+                if self.device == "cpu":
+                    self.dtype = self.output_tensor_list[i].dtype()
+                    self.output = np.empty(self.dimensions, dtype=self.dtype)
+                    self.labels = np.empty(self.labels_size, dtype=self.dtype)
+                else:
+                    self.dtype = self.output_tensor_list[i].dtype()
+                    with cp.cuda.Device(device=self.device_id):
+                        self.output = cp.empty(self.dimensions, dtype=self.dtype)
+                        self.labels = cp.empty(self.labels_size, dtype=self.dtype)
+
+                if self.device == "cpu":
+                    self.output_tensor_list[i].copy_data_numpy(self.output)
+                else:
+                    self.output_tensor_list[i].copy_data_cupy(self.output.data.ptr)
+                self.output_list.append(self.output)
+        else:
+            for i in range(len(self.output_tensor_list)):
+                if self.device == "cpu":
+                    self.output_tensor_list[i].copy_data_numpy(self.output_list[i])
+                else:
+                    self.output_tensor_list[i].copy_data_cupy(self.output_list[i].data.ptr)
+
+        if (self.loader._name == "labelReader"):
+            if (self.loader._one_hot_encoding == True):
+                self.loader.getOneHotEncodedLabels(self.labels, self.device)
+                self.labels_tensor = self.labels.reshape(-1, self.batch_size, self.loader._num_classes)
             else:
                 if self.display:
-                    for i in range(self.bs):
-                        img = (self.out)
-                        draw_patches(img[i], i, 0)
-                self.labels = self.loader.rocalGetImageLabels()
+                    for output in self.output_tensor_list:
+                        for i in range(self.batch_size):
+                            draw_patches(output[i], i, 0)
+                self.labels = self.loader.getImageLabels()
                 if self.device == "cpu":
-                    self.labels_tensor = self.labels.astype(dtype = np.int_)
+                    self.labels_tensor = self.labels.astype(dtype=np.int_)
                 else:
-                    with cp.cuda.Device(device = self.device_id):
-                        self.labels_tensor = self.labels.astype(dtype = cp.int_)
+                    with cp.cuda.Device(device=self.device_id):
+                        self.labels_tensor = self.labels.astype(dtype=cp.int_)
 
-            return self.out, self.labels_tensor
+        return self.output_list, self.labels_tensor
```

Comment:
Have we tested id returning as the lists works for the trainings ?

Reply:
I'm using list unpacking in classification and other training examples to return the single output. Like this
for ( [img], target) in rocal_iterator: